pkg/collector/corechecks/ebpf/probe/ebpfcheck/probe.go

```diff
@@ -214,13 +215,20 @@ func (k *Probe) getProgramStats(stats *model.EBPFStats) error {
 	var err error
 	progid := ebpf.ProgramID(0)
 	for progid, err = ebpf.ProgramGetNextID(progid); err == nil; progid, err = ebpf.ProgramGetNextID(progid) {
-		pr, err := ebpf.NewProgramFromID(progid)
+		fd, err := ProgGetFdByID(&ProgGetFdByIDAttr{ID: uint32(progid)})
 		if err != nil {
-			log.Debugf("unable to get program prog_id=%d: %s", progid, err)
+			log.Debugf("error getting program info prog_id=%d: %s", progid, err)
```

Comment:
nit: getting the fd, not the info
```suggestion
			log.Debugf("error getting program fd prog_id=%d: %s", progid, err)
```